### PR TITLE
Cesium Viewer flyTo query parameter

### PR DESCRIPTION
**Pull request by pjcozzi**
Do we want this (the idea; the code could be better I'm sure)?

This is useful for testing and will be good for tweets like [these](https://twitter.com/CesiumJS/status/567698151528615938) that we want to start doing more of.

I'm not married to this exact approach, e.g., if we wanted to change the parameters or embed CZML or something in the url.

### CODE REVIEW

**Comment by mramato** [comment]
We've always talked about adding it, but it all depends on how much work we want to do specifically for the `Cesium Viewer` application itself.  

If we want to do it, I would probably use a fragment,`#`, instead and also update it every time the camera moves, this way you can email someone a link.  This is similar to how all the other web maps work, like Google Maps for instance: https://www.google.com/maps

**Comment by pjcozzi** [comment]
> If we want to do it, I would probably use a fragment,#, instead and also update it every time the camera moves, this way you can email someone a link.

Sounds good.  I wonder if there is a performance hit?

**Comment by mramato** [comment]
Depends on how you implement it.  I would probably add that view changed event we talked about when writing the GE demos and use that and only update it when the camera is done moving.  At that point I think there would be zero impact.

**Comment by pjcozzi** [comment]
Sounds cool.  Perhaps we leave this open for when we go on tour next?

**Comment by emackey** [comment]
+1 for this.

**Comment by hpinkos** [comment]
@pjcozzi this should be ready

**Comment by pjcozzi** [comment]
I don't think the camera move event is fired often event.  For example, run this:
```
http://localhost:8080/Apps/CesiumViewer/index.html?view=-95.00000000000001%2C35.121308333162105%2C2497.57369106689%2C360%2C-89.81910938407604%2C0
```
Then keep zooming in and the url never updates.

@bagnell?

**Comment by pjcozzi** [comment]
@emackey I'm still interested in the Sandcastle question from the other PR.

**Comment by pjcozzi** [comment]
> If we want to do it, I would probably use a fragment,#...

Just saw this from @mramato.  Whatever you guys think.

**Comment by hpinkos** [comment]
> If we want to do it, I would probably use a fragment,#...

I briefly looked into this earlier, and I really don't think it makes a difference either way.

**Comment by emackey** [comment]
>>If we want to do it, I would probably use a fragment,#...

>I briefly looked into this earlier, and I really don't think it makes a difference either way.

Not much difference, but some link-sharing sites (Reddit) don't allow the same link to be re-shared with only the hash (#) being different.  So keeping it in the queryString might be better.  Looks like Google Maps actually embeds it directly in the path now.

@pjcozzi I'll think more about the Sandcastle issue but that should be a separate PR from this.

**Comment by mramato** [comment]
The difference between a query and a hash is that hashes aren't sent to the server.  In this case it doesn't matter (and because of @emackey's point about some social media sites), leaving it as a query is probably fine.  In reality Cesium Viewer is just a demo app with no guaranteed functionality, so we can change all of this at our whim.

Google Maps is probably using push state, which is what a real Cesium app would probably want as well, but that's overkill for the viewer.

**Comment by pjcozzi** [comment]
This is just waiting on the [camera update issue](https://github.com/AnalyticalGraphicsInc/cesium/pull/2500#issuecomment-102188520).

@bagnell can you advise?

**Comment by bagnell** [comment]
You can lower the epsilon value [here](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/Scene.js#L1401). I was trying to come up with a good value that would work with large positions and the orientation without testing them all separately. Perhaps `EPSILON6` is a good value?

**Comment by emackey** [comment]
@bagnell the problem is that `camera.position` needs a radically different epsilon from direction, up, right, and the transform.  This is because those other values are all normalized, where `position` is not.

**Comment by hpinkos** [comment]
@bagnell increasing the epsilon to 8 fixed the problem @pjcozzi found with zooming close into the globe.
But i'm pretty sure it make #2715 worse.

Is there something else I should do? Based on what @emackey said, it looks like we might need a better fix.

**Comment by emackey** [comment]
Actually I have an idea... let me test it

**Comment by bagnell** [comment]
@emackey may be right and we need to have different epsilon tests. We would definitely need to change it for absolute error tests, but the `equalsEpsilon` functions use a relative error test as well which I think would be fine for both the position and orientation.

**Comment by emackey** [comment]
OK, let me know what you think of this.  I modified the equality check to scale the position by the longest dimension (kind of like normalization but without the squares and square roots).  Also I fixed the issue with the flickering start/end events.  And I put in a fix for when tiny below-epsilon camera movements eventually add up to a significant movement (which is common near the surface).

But, there's a new issue where it decides the cam is done moving, but tiny amounts of inertia are still going, and that bit of leftover inertia is eventually able to trigger a new move event by adding itself up over a couple of frames.  This results in one tiny "bounce" (start/end event pair) at the end of a large move.  This isn't as bad as the flickering, IMHO.  If people care, this could be fixed by having the inertia turn itself off when a move-end event fires.

**Comment by pjcozzi** [comment]
Can we aim to get this merged today to wrap up items from the sprint?

@bagnell can you look at @emackey's change?

**Comment by bagnell** [comment]
@emackey's change looks good to me. You should remove `Camera.equalsEpsilon` and make it a file scoped function in `Scene`.

**Comment by emackey** [comment]
I won't be able to get to this until later tonight.  @hpinkos are you up for moving the function @bagnell mentioned?

**Comment by hpinkos** [comment]
@emackey no problem.  @bagnell ready

**Comment by bagnell** [comment]
Looks good to me. @pjcozzi Do you want to merge?

File: Apps/CesiumViewer/CesiumViewer.js [hunk 1]
```diff
@@ -34,6 +40,9 @@ define([
      * 'stats'  : true,         // Enable the FPS performance display.
      * 'theme'  : 'lighter',    // Use the dark-text-on-light-background theme.
      * 'scene3DOnly' : false    // Enable 3D only mode
+     * 'view' : longitude,latitude,[height,heading,pitch,roll]
```
Comment (pjcozzi) [inline Apps/CesiumViewer/CesiumViewer.js]: Can we update CHANGES.md and include an example of this?

**Comment by pjcozzi** [comment]
This is good, just update CHANGES.md.

**Comment by hpinkos** [comment]
@pjcozzi updated CHANGES.md

**Pull request merged**